Sources/XcodeGenKit/SettingsBuilder.swift

```diff
@@ -57,15 +57,19 @@ extension ProjectSpec {
         return buildSettings
     }
 
-    // combines all levels of a target's settings
+    // combines all levels of a target's settings: target, target config, project, project config
     public func getCombinedBuildSettings(basePath: Path, target: Target, config: Config, includeProject: Bool) -> BuildSettings {
         var buildSettings: BuildSettings = [:]
         if includeProject {
+            if let configFilePath = configFiles[config.name] {
+                if let configFile = try? XCConfig(path: basePath + configFilePath) {
+                    buildSettings += configFile.flattenedBuildSettings()
```

Comment:
Maybe including xcconfig should just `#include` the included xcconfig, not adding to buildSettings section.🤔
What do you think?

Reply:
Oh and what if included xcconfig `#include` other xcconfigs?

Reply:
Not sure what you mean by the first point? Maybe these will answer your question.
- this function is just use for getting all the combined settings for the purposes of checking what build settings will be applied - similar to Xcode's `combined` settings view
- xcconfigs aren't added using #include they are just set as a baseconfiguration for the target or project. The rest of the build settings are applied directly on the build setting or project and don't use an xcconfig

For the second point `configFile. flattenedBuildSettings()` goes through all #include files and merges them 👍

Reply:
Understood. I'm going to take a look at the details.👌

Reply:
> For the second point configFile. flattenedBuildSettings() goes through all #include files and merges them 👍

What I meant was `XcodeGen:Include`, I but I see `XcodeGen:Include` just ignores `XcodeGen:Include`ed spec's `configFile`, so it's no problem.

```
TestProject $ head spec.yml
include: included.yml
name: GeneratedProject
groups:
  - Configs
targets:
  TestProject:
    type: application
    platform: iOS
    sources: TestProject
    settings:
TestProject $ head included.yml
configFiles:
  Debug: Included/Configs/included.xcconfig  # This is completely ignored when `XcodeGen:included`.
```
This behavior is perfectly 🙆‍♂️ to me. Maybe it worth it to warn user when `XcodeGen:included` spec has configFile attribute, because it's ignored.

Maybe we should really rename `XcodeGen:include` to something else to avoid confusion. 🙄

Reply:
That's because include is a list not just a string. I should change this though to allow a simple string for a single value, just like a target's sources:
```
include: [included.yml] #works, using json style
include:
  - included.yml #works, using yaml style
include: included.yml #does work (yet, will change)
```

Reply:
`include` can now be a single string https://github.com/yonaskolb/XcodeGen/commit/39c8af324c88f0e43f7131c5791f718a1eb29de6